app/models/concerns/spotlight/solr_document/active_model_concern.rb

```diff
@@ -10,7 +10,7 @@ module Spotlight::SolrDocument::ActiveModelConcern
   module ClassMethods
 
     def primary_key
-      :id
+      self.unique_key
```

Comment:
I'm actually a little surprised at this change. You need to do this even after configuring your `SolrDocument` correctly by telling it what the unique key column is?

https://github.com/projectblacklight/blacklight/blob/master/lib/blacklight/solr/document.rb#L174-177